examples/ts_js_interop/BUILD.bazel

```diff
@@ -0,0 +1,25 @@
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+load("@aspect_rules_swc//swc:defs.bzl", "swc")
+load("@bazel_skylib//lib:partial.bzl", "partial")
+
+ts_project(
+    name = "src",
+    srcs = [
+        "a.js",
+        "b.ts",
+    ],
+    allow_js = True,
+    declaration = True,
+    emit_declaration_only = True,
```

Comment:
Commenting this out causes the `b.js` to be generated, but also causes a conflict when building all targets under this directory.

Reply:
Lets comment it out then to reproduce the bug?

Reply:
Both commented and uncommented are bugs.

Uncommented: I expect this example to build, not fail to produce JavaScript therefore not being able to create the file group.
Commented: I at least expect the targets to build, not result in that conflict error? Are you able to see both errors on your end when pulling these changes?

Reply:
In other words, I'm between a rock and a hard place. If I don't emit, I don't get any transpiled JavaScript, even though I need the js to pass further downstream. In this example, I have reproduced that with the file group rule. 

If I do emit, I get an error about the same js file being produced by the transpile and type check steps.

My current workaround is to produce my own ts_project macro which calls tsc to produce the .d.ts files and calls swc to produce the transpiles. I can still use ts_project if all sources are typescript.

Reply:
Can we have 2 `ts_project` targets (maybe with different srcs) to reproduce both of those bugs?

I mainly just want the minimum code to reproduce the issue(s) without any extra code mixed any causing confusion.

Reply:
Ok, added a second `ts_project` target